lab/skel_graph/traversal/problems/alee/alee.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 Nicolae Mara-Ioana <maraioana9967@gmail.com>
+
+// Alee - Drumuri minime in graf
+// Complexitate: O(n * m)
+
+// https://www.infoarena.ro/problema/alee
+
+#include <bits/stdc++.h>
+
+class Task {
+	
+ public:
+	
+	void solve() {
+		read_input();
+		print_output(get_result());
+	
+	}
+
+ private:
+	int n;
+	int m;
+	int source;
+	std::vector<std::vector<char>> graph;
+    std::pair<int, int> startGate;
+    std::pair<int, int> endGate;
+	
+    // s-a ales urmatoarea codificare:
+    // graph[i][j] = 'T', atunci exista un copcac la coordonatele i, j
+    // graph[i][j] = ' ', atunci exista zona libera la coordonatele i, j
+	void read_input() {	
+        std::cin >> n >> m;
+        // initial se marcheaza toate elementele din matrice ca fiind zone libere
+        graph = std::vector<std::vector<char>>(n + 1, 
+            std::vector<char>(n + 1, ' '));
+		for (int i = 1, x, y; i <= m; i++) {
+            // se citesc coordonatele la care se afla copacii
+            // se actualizeaza valorile din celulele in care se regasesc obstacole
+			std::cin >> x >> y;
+			graph[x][y] = 'T';
```

Comment:
I think it's may be a bit complicated to have 'T' and ' '. Maybe change to boolean and have 1 for tree and 0 for no tree?